src/pkg/services/m365/api/channels.go

```diff
@@ -1 +1,376 @@
+func (c Channels) DeleteChannel(
+	ctx context.Context,
+	teamID, containerID string,
+) error {
+	// deletes require unique http clients
+	// https://github.com/alcionai/corso/issues/2707
+	srv, err := NewService(c.Credentials)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	err = srv.Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Delete(ctx, nil)
+	if err != nil {
+		return graph.Stack(ctx, err)
+	}
+
+	return nil
+}
+
+// prefer GetChannelByID where possible.
+// use this only in cases where the models.Channelable
+// is required.
+func (c Channels) GetChannel(
+	ctx context.Context,
+	teamID, containerID string,
+) (models.Channelable, error) {
+	config := &teams.ItemChannelsChannelItemRequestBuilderGetRequestConfiguration{
+		QueryParameters: &teams.ItemChannelsChannelItemRequestBuilderGetQueryParameters{
+			Select: idAnd("name", "owner"),
+		},
+	}
+
+	resp, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Get(ctx, config)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return resp, nil
+}
+
+func (c Channels) GetChannelByID(
+	ctx context.Context,
+	teamID, containerID string,
+) (graph.Container, error) {
+	channel, err := c.GetChannel(ctx, teamID, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	return ChannelsDisplayable{Channelable: channel}, nil
+}
+
+// GetChannelByName fetches a calendar by name
+func (c Channels) GetChannelByName(
+	ctx context.Context,
+	teamID, _, containerName string,
+) (graph.Container, error) {
+
+	ctx = clues.Add(ctx, "channel_name", containerName)
+
+	resp, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		Get(ctx, nil)
+
+	if err != nil {
+		return nil, graph.Stack(ctx, err).WithClues(ctx)
+	}
+
+	gv := resp.GetValue()
+
+	if len(gv) == 0 {
+		return nil, clues.New("channel not found").WithClues(ctx)
+	}
+
+	// We only allow the api to match one calendar with the provided name.
+	// If we match multiples, we'll eagerly return the first one.
+	logger.Ctx(ctx).Debugw("calendars matched the name search", "calendar_count", len(gv))
+
+	// Sanity check ID and name
+	cal := gv[0]
+	container := ChannelsDisplayable{Channelable: cal}
+
+	if err := graph.CheckIDAndName(container); err != nil {
+		return nil, clues.Stack(err).WithClues(ctx)
+	}
+
+	return container, nil
+}
+
+func (c Channels) PatchChannel(
+	ctx context.Context,
+	teamID, containerID string,
+	body models.Channelable,
+) error {
+	_, err := c.Stable.
+		Client().
+		Teams().
+		ByTeamId(teamID).
+		Channels().
+		ByChannelId(containerID).
+		Patch(ctx, body, nil)
+
+	if err != nil {
+		return graph.Wrap(ctx, err, "patching event calendar")
+	}
+
+	return nil
+}
+
+// ---------------------------------------------------------------------------
+// message
+// ---------------------------------------------------------------------------
+
+// GetItem retrieves a Messageable item.
```

Comment:
Comments here are out of sync.  Please go through the file and make sure they match the data being used.